org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/entity/trigger/AbstractManagedElementTriggerChecker.java

```diff
@@ -58,7 +63,8 @@ public AbstractManagedElementTriggerChecker(final BaseTrigger trigger,
 		this.managedElementsStateStimulus = stimulus;
 		this.metricSetDescription = metricSetDescription;
 		this.baseMetricDescription = baseMetricDescription;
-		this.aggregator = FixedLengthWindowAggregation.getFromAggregationMethod(stimulus.getAggregationOverElements());
+//		this.aggregator = FixedLengthWindowAggregation.getFromAggregationMethod(stimulus.getAggregationOverElements());
+		this.aggregator = new SlidingTimeWindowAggregationBasedOnEMA(60,10,0.2);
```

Comment:
iirc, we discussed this last week and the conclusion was this:
* based on SPD model, we cannot decide which Type of Window to use. 
  * for comparison, monitoring models contain information about what type of aggrgation to use and so on but this is tiresome and not always relevant for the modeller, i.e. we do not want this type of information in the SPD model.
* However, it would be a waste to let more efficient Windwos, such as the EMA based window, that is only for AVG, go to waste. 
* **Solution** : make the Type of window a Slingshot-Design-Time decission, i.e. the Slingshot Developpers (us) decide, which window to use in which case, with a simple if-else, or a switch-case, or something like that. In this case it would be, *if the aggregation is AVG, use the EMA based aggregation, in all other cases fall back to default aggregation window.*
  * In case someone implements a new aggregator, that is more efficient that the fallback / default, the code base must be extended to include the new aggregator.